packages/peregrine/lib/talons/SearchPage/__tests__/useSearchPage.spec.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { act } from 'react-test-renderer';
+
+import { useSearchPage } from '../useSearchPage';
+import createTestInstance from '../../../util/createTestInstance';
+
+jest.mock('react-router-dom', () => ({
+    useHistory: jest.fn(() => ({ push: jest.fn() })),
+    useLocation: jest.fn(() => ({ pathname: '', search: '' }))
+}));
+
+jest.mock('../../../context/app', () => {
+    const state = {};
+    const api = {
+        toggleDrawer: jest.fn()
+    };
+    const useAppContext = jest.fn(() => [state, api]);
+
+    return { useAppContext };
+});
+
+const mockUseSort = jest
+    .fn()
+    .mockReturnValue([
+        {
+            sortText: 'Best Match',
+            sortAttribute: 'relevance',
+            sortDirection: 'DESC'
+        },
+        jest.fn()
+    ])
+    .mockName('mockUseSort');
+
+const mockSetCurrentPage = jest.fn().mockName('mockSetCurrentPage');
+
+jest.mock('@magento/peregrine', () => {
+    const usePagination = jest.fn(() => [
+        {
+            currentPage: 3,
+            totalPages: 6
+        },
+        {
+            setCurrentPage: jest
+                .fn()
+                .mockImplementation(() => mockSetCurrentPage()),
+            setTotalPages: jest.fn()
+        }
+    ]);
+    const useSort = jest.fn().mockImplementation(() => mockUseSort());
+
+    return {
+        useSort,
+        usePagination
+    };
+});
+
+jest.mock('@apollo/react-hooks', () => {
+    const useQuery = jest.fn().mockReturnValue({
+        data: {
+            __type: {
+                inputFields: []
+            }
+        },
+        error: null,
+        loading: false
+    });
+    const runQuery = jest.fn();
+    const queryResult = {
+        data: {
+            products: {
+                page_info: {
+                    total_pages: 6
+                }
+            }
+        },
+        error: null,
+        loading: false
+    };
+    const useLazyQuery = jest.fn(() => [runQuery, queryResult]);
+
+    return { runQuery, queryResult, useLazyQuery, useQuery };
```

Comment:
Not causing any harm, but these shouldn't be in the return.

```suggestion
    return { useLazyQuery, useQuery };
```

Reply:
updated